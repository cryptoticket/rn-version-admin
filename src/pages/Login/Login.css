.login {
	display: flex;
	justify-content: center;
	margin-top: 15%;
}

.login__card {
	width: 320px;
}

.login input {
	width: 100%;
}
