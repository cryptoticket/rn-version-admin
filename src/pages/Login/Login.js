import { Button } from 'primereact/button';
import { Card } from 'primereact/card';
import React from 'react';
import './Login.css';

/**
 * Component renders a login page
 */
export default class Login extends React.Component {

	/**
	 * On button "login" click
	 */
	onLoginClick = () => {
		// if gate (3rd party oauth service) url is set then redirect to gate
		if(process.env['REACT_APP_OAUTH_GATE_URL']) {
			window.location.href = process.env['REACT_APP_OAUTH_GATE_URL'];
		} else {
			// redirect to google oauth by default
			window.location.href = '/api/v1/auth/google';
		}
	};

	/**
	 * Renders JSX template
	 * @return {Object} JSX template
	 */
	render() {
		return (
			<div className="login">
				<Card className="login__card">
					<div className="p-inputgroup p-justify-center">
						<Button label="Login with Google" icon="pi pi-google" onClick={this.onLoginClick} />
					</div>
				</Card>
			</div>
		);
	}
}
