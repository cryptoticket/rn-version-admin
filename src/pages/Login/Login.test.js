import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';

import Login from './Login';

configure({ adapter: new Adapter() });

describe('Login component', () => {
    describe('onLoginClick()', () => {
		it('should redirect user to google auth page', function () {
			delete window.location;
			window.location = {};
			const wrapper = shallow(<Login />);
			const instance = wrapper.instance();
			instance.onLoginClick();
			expect(window.location.href).toEqual('/api/v1/auth/google');
		});
	});

	describe('render()', () => {
		it('should render component', function () {
			const component = renderer.create(<Login />);
			expect(component.toJSON()).toMatchSnapshot();
		});
	});
});
