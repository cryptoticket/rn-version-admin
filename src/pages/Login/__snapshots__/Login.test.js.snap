// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component render() should render component 1`] = `
<div
  className="login"
>
  <div
    className="p-card p-component login__card"
    style={null}
  >
    <div
      className="p-card-body"
    >
      <div
        className="p-card-content"
      >
         
        <div
          className="p-inputgroup p-justify-center"
        >
          <button
            className="p-button p-component p-button-text-icon-left"
            onClick={[Function]}
          >
            <span
              className="pi pi-google p-c p-button-icon-left"
            />
            <span
              className="p-button-text p-c"
            >
              Login with Google
            </span>
          </button>
        </div>
         
      </div>
    </div>
  </div>
</div>
`;
